controller/pb/controller.proto

```diff
@@ -25,32 +25,54 @@ message BuildOptions {
   string ContextPath = 1;
   string DockerfileName = 2;
   string PrintFunc = 3;
-
-  repeated string Allow = 4;
-  repeated string Attests = 5; // TODO
-  repeated string BuildArgs = 6;
-  repeated string CacheFrom = 7;
-  repeated string CacheTo = 8;
-  string CgroupParent = 9;
-  repeated string Contexts = 10;
-  repeated string ExtraHosts = 11;
-  string ImageIDFile = 12;
-  repeated string Labels = 13;
-  string NetworkMode = 14;
-  repeated string NoCacheFilter = 15;
-  repeated string Outputs = 16;
+  map<string, string> NamedContexts = 4;
```

Comment:
We should try to group the things that belong together. Either in substructs where they make sense or at least reserved ID ranges.

Currently, everything is randomly mixed up. Parameters sent to frontend, input contexts with print flags, imageID/quiet that are client-side options(quiet probably shouldn't be here at all. Nice if things that rely on session would be separate as well. CommonOpts is not a good name and I don't understand the property split there.

Reply:
> We should try to group the things that belong together. Either in substructs where they make sense or at least reserved ID ranges.

Agreed. Can we do this as a follow-up, so as to not block https://github.com/docker/buildx/pull/1581? Substructs would be my preference wherever possible.

> imageID/quiet that are client-side options(quiet probably shouldn't be here at all)

Have pushed a commit that removes the quiet option entirely from the controller api - we can also extract the imageID file writing from the controller as well as a follow-up. To do this, we have the api return the solve response of the build, so we can extract the image id (which is needed because `--quiet` prints the image id, functionality that was previously broken on the remote controller).

> CommonOpts is not a good name and I don't understand the property split there.

Currently, this is because these are command line options and are entirely shared with bake. This is another thing to refactor, but it's kind of all rolled up into "how do we get bake calling the controller api" - would prefer to address this a bit later, but it's on my todo list!